app/forms/delivery_partner_form.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+class DeliveryPartnerForm
+  include ActiveModel::Model
+
+  attr_accessor :name, :lead_providers, :provider_relationships
+  validate :lead_providers_and_cohorts_validation
+
+  def available_lead_providers
+    LeadProvider.includes(:cohorts).select { |lead_provider| lead_provider.cohorts.any? }
+  end
+
+  def chosen_provider_relationships
+    provider_relationships.map do |relationship|
+      ProviderRelationship.new(
+        cohort: Cohort.find(relationship["cohort_id"]),
+        lead_provider: LeadProvider.find(relationship["lead_provider_id"]),
+      )
+    end
+  end
+
+  def display_lead_provider_details
+    chosen_provider_relationships.group_by(&:lead_provider).map do |lps_to_relationships|
+      {
+        name: lps_to_relationships.first.name,
+        chosen_cohorts: lps_to_relationships.second.map { |relationship| relationship.cohort.display_name }.join(", "),
+      }
+    end
+  end
+
+  def populate_provider_relationships(params)
+    self.lead_providers = []
+    self.provider_relationships = []
+
+    lead_provider_ids = params.dig(:delivery_partner_form, :lead_providers)&.keep_if(&:present?)
+    lead_provider_ids.each do |lead_provider_id|
+      lead_providers.push(LeadProvider.find(lead_provider_id))
```

Comment:
You can just load / asiign all your lead providers by passing in the collection of ids LeadProvider.find(:lead_provider_ids).

Reply:
Thanks!